app/Http/Controllers/Api/PublicScheduleController.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use App\Http\Controllers\Controller;
+use App\Models\Lecturing;
+use Carbon\Carbon;
+use Illuminate\Http\Request;
+
+class PublicScheduleController extends Controller
+{
+    public function index(Request $request)
+    {
+        $lecturingQuery = Lecturing::query();
+        $startDate = $request->startDate ? $request->startDate : Carbon::now()->startOfMonth();
+        $endDate = $request->endDate ? $request->endDate : Carbon::now()->endOfMonth();
```

Comment:
Ada beberapa koreksi mas, agar seragam dengan konvensi kode di proyek ini
1. Untuk query string, key nya pakai snake_case: `start_date`.
2. Untuk mengambil payload request, kita pakai `$request->get('start_date')`. Tidak `$request->startDate`
3. Dengan `$request->get()`, kita bisa kasih default value di parameter kedua, seperti: `$request->get('start_date', 'default_value')`
4. Karena kita butuh tanggal saja, maka Carbonnya langsung kita kasih format `->format('Y-m-d');`

```suggestion
        $startDate = $request->get('start_date', Carbon::now()->startOfMonth()->format('Y-m-d'));
        $endDate = $request->get('end_date', Carbon::now()->endOfMonth()->format('Y-m-d'));
```